extend view function to accomodate node/number args, generate constraints for scalar addition and lt

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81360
* #81344
* #81274
* #81265
* #81190
* #81189
* #81175
* #81159
* #80976
* #80925
* #80909
* #80847
* __->__ #80823
* #80799
* #80744


- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

migeed-z committed:
extend view function to accomodate node/number args, generate constraints for scalar addition and lt

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

jamesr66a reviewed (commented):
Can you add a description of the functional changes in the PR desciption?

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"




[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z committed:
Update on "extend view function to accomodate node/number args, generate constraints for scalar addition and lt"



- View function can contain arguments that are nodes or numbers. Before, we were assuming all the arguments are numbers which is not general. We modify the code to handle this case. Some of the other inference rules will also be updated accordingly. 

- Generate scalar addition constraints of the form input = output 
- Generate lt constraints which are the same as those for gt.

[ghstack-poisoned]

migeed-z commented:
@pytorchbot merge -g

migeed-z commented:
@pytorchbot merge -g
